src/extensions/default/PhpTooling/CodeHintsProvider.js

```diff
@@ -0,0 +1,205 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/* eslint-disable indent */
+/* eslint max-len: ["error", { "code": 200 }]*/
+define(function (require, exports, module) {
+    "use strict";
+
+    var _ = brackets.getModule("thirdparty/lodash");
+
+    var EditorManager = brackets.getModule('editor/EditorManager'),
+        ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
+        TokenUtils = brackets.getModule("utils/TokenUtils"),
+        StringMatch = brackets.getModule("utils/StringMatch"),
+        matcher = new StringMatch.StringMatcher({
+            preferPrefixMatches: true
+        });
+
+    ExtensionUtils.loadStyleSheet(module, "../../../languageTools/styles/default_provider_style.css");
+    var phpSuperGlobalVariables = JSON.parse(require("text!phpGlobals.json"));
+
+    function CodeHintsProvider(client) {
+        this.client = client;
+        this.query = "";
+    }
+
+    function formatTypeDataForToken($hintObj, token) {
+        $hintObj.addClass('brackets-hints-with-type-details');
+        if (token.detail) {
+            if (token.detail.trim() !== '?') {
+                if (token.detail.length < 30) {
+                    $('<span>' + token.detail.split('->').join(':').toString().trim() + '</span>').appendTo($hintObj).addClass("brackets-hints-type-details");
+                }
+                $('<span>' + token.detail.split('->').join(':').toString().trim() + '</span>').appendTo($hintObj).addClass("hint-description");
+            }
+        } else {
+            if (token.keyword) {
+                $('<span>keyword</span>').appendTo($hintObj).addClass("brackets-hints-keyword");
+            }
+        }
+        if (token.documentation) {
+            $hintObj.attr('title', token.documentation);
+            $('<span></span>').text(token.documentation.trim()).appendTo($hintObj).addClass("hint-doc");
+        }
+    }
+
+    function filterWithQueryAndMatcher(hints, query) {
+        var matchResults = $.map(hints, function (hint) {
+            var searchResult = matcher.match(hint.label, query);
+            if (searchResult) {
+                for (var key in hint) {
+                    searchResult[key] = hint[key];
+                }
+            }
+
+            return searchResult;
+        });
+
+        return matchResults;
+    }
+
+    CodeHintsProvider.prototype.hasHints = function (editor, implicitChar) {
+        if (!this.client) {
+            return false;
+        }
+
+        return true;
+    };
+
+    CodeHintsProvider.prototype.getHints = function (implicitChar) {
```

Comment:
I don't think we need this file. You can create override the getHInts function of the DefaultProvider object since that is the only place is where you need to change anything. See my comments below.

Reply:
see my comment below